source/going_further.rst

```diff
@@ -0,0 +1,472 @@
+helpful in your day-to-day python programming.
+
+
+Managing larger projects
+========================
+
+Divide and conquer
+--------------------
+
+Up to this point, you might have put all of your source coed in one file. For small libraries and project this is fine, but when complexity
+rises it is better to divide your code into smaller files, that is to create libraries or modules. It is not necessary, for example, to put
+every class in a separate file, but having one file for classes that are logically connected is better for readability.
+Going through a file that has thousands lines of code can be difficult.
+
+As an example, let us consider a program for handling regular polygons. Imagine that every figure has its own, simplified class. You might want to
+consider separating main functionality e. g. calculating perimeter, from definitions of those shapes [#fn1]_. That is, to put calculations of fields in one
+file and definitions in another:
+
+.. code-block:: python
+    
+    # File polygons.py
+    class Circle:
+        def __init__(self, radius):
+            self.no_sides = 0
+            self.radius = radius
+
+    class Triangle:
+        def __init__(self, side):
+            self.no_sides = 3
+            self.side = side
+
+    class Square:
+        def __init__(self):
+            self.no_sides = 4
+            self.side = side
+
+.. code-block:: python
+
+    # File shapefunctions.py
+    PI = 3.141592 # or use math.PI
+    def calc_perimeter(obj):
+        if obj.no_sides > 0:
+            perimeter = obj.no_sides * obj.side
+        else:  
+            perimeter = 2*PI*obj.radius
+        return perimeter
+
+Importing functionalities
+-------------------------
+
+To use definitions and perimeter function you must now import those functionalities. Code below asks user to specify which shape to
+use and for the side length/radius, and calculates perimeter:
+
+.. code-block:: python
+
+    # Main file or command line
+    import polygons
+    import shapefunctions
+
+    print("Choose a shape:")
+    print("1. Circle")
+    print("2. Triangle")
+    print("3. Square")
+    choice = int(input())
+    print("Choose the side length")
+    side = int(input())
+    if choice == 1:
+        shape = polygons.Circle(side)
+    elif choice == 2:
+        shape = polygons.Triangle(side)
+    elif choice == 3:
+        shape = polygons.Square(side)
+    per = shapefunctions.perimeter(shape)
+    print("The perimeter of your shape is %3.3f" % per)
+
+The ``import polygons`` and ``import shapefunctions`` lines tell the program that we are going to use functionality defined in files ``polygons.py`` and ``shapefunctions.py``.
+Note that these files must be in a place Python can find them, e.g. in the same directory you are running
+python from or on a path defined by ``PYTHONPATH`` variable. Also, note that in the imports we are not using file type extensions, i.e. there is no ``.py`` in
```

Comment:
This statement reads as if those are the two equally popular ways of placing files. Maybe consider adding a sentence or two describing when it's recommended to use which way?